components/locale/examples/langid.rs

```diff
@@ -0,0 +1,43 @@
+// Simple app
+use icu_locale::{subtags, LanguageIdentifier};
+
+const LANGIDS: &[&str] = &[
+    "de",
+    "en-US",
+    "zh-Hant",
+    "sr-Cyrl",
+    "fr-CA",
+    "es-CL",
+    "pl",
+    "en-Latn-US",
+    "ca-valencia",
+    "und-Arab",
+];
+
+fn main() {
+    // 1. Parse input strings into LanguageIdentifiers
+    let langids = LANGIDS
+        .iter()
+        .map(|s| s.parse())
+        .collect::<Result<Vec<LanguageIdentifier>, _>>()
+        .expect("Failed to parse Language Identifier");
+
+    {
+        // 2. Filter list of identifiers that match a given Language Identifier.
+        let reference: LanguageIdentifier = "en-US"
```

Comment:
Use your macro here instead?  There should no longer be any reason to hard-code a `"...".parse().expect("...")` pattern for LanguageIdentifier.

Reply:
Once we merge the macros! :)

Reply:
I think it would be good to use the macros in this example.  Hopefully the macros will be checked in soon.